slackbot-background/utils.go

```diff
@@ -44,27 +53,48 @@ func checkUsersGroups(email string) (bool, error) {
 	return false, nil
 }
 
-func parseCommandText(text string) (net.IP, time.Time, string, error) {
+func parseWhitelistText(text, typestr string) (string, time.Time, string, error) {
 	splitCmd := strings.Split(text, " ")
 
-	ip := net.ParseIP(splitCmd[0])
-	if ip == nil {
-		m := fmt.Sprintf("Got invalid IP: %s", splitCmd[0])
+	obj := splitCmd[0]
+	var err error
+	if typestr == common.EMAIL_TYPE {
+		err = isEmailValid(obj)
+	}
+	if typestr == common.IP_TYPE {
+		ip := net.ParseIP(obj)
+		if ip == nil {
+			err = fmt.Errorf("Got invalid IP: %s", obj)
+		} else {
+			obj = ip.String()
+		}
+	}
```

Comment:
similar to previous comment, if the above blocks don't match `err` will be non-nil and we will proceed past the block below